### DISCUSSION TITLE
[CLI] New Certificate Management Commands

### DISCUSSION THREAD

**Discussion by sintaxi** [sintaxi/surge#446] [Show and tell]
Two new commands (`certs` & `encrypt`) have landed CLI for Managing Certificates. These commands are currently in the `edge` release of surge and will be included in the next release of surge.

## `surge certs <domain>`

This command returns all certs that are detected on our system. This includes both uploaded certs and certs that are provisioned by the surge platform via Lets Encrypt. This command shows the **certificate authority**, when the cert will **expire**, and if the cert is set to **auto-renew** by the surge platform. This list represents the source of truth for certificates on the surge platform.

Here is an example running `surge certs sintaxi.com` where a certificate is about to expire...

<img width="1123" alt="Screen Shot 2021-04-04 at 12 10 29 PM" src="https://user-images.githubusercontent.com/867/113519269-f0c73f00-953f-11eb-89a4-ee867f558f14.png">

## `surge encrypt <domain>`

This command provisions a new cert if there is not a fresh certificate already existing on the platform. Those on the **Student** plan will be prompted to upgrade to the **Professional** plan - as this is a paid feature. The Professional plan permits provisioning certs on an unlimited number of domains.

Here is an example running `surge encrypt sintaxi.com`. As you can see a new cert is provisioned for the domain...

<img width="1123" alt="Screen Shot 2021-04-04 at 12 12 37 PM" src="https://user-images.githubusercontent.com/867/113519559-e7d76d00-9541-11eb-8239-52fc82617504.png">

Running `surge certs sintaxi.com` again confirms the new certificate is on the system and valid for 90 days...

<img width="1123" alt="Screen Shot 2021-04-04 at 12 12 58 PM" src="https://user-images.githubusercontent.com/867/113519617-400e6f00-9542-11eb-9744-88e566931abf.png">

These two commands are available for preview now in the edge branch of surge. It can be installed by running `npm install -g surge@edge`. [Please open an issue if any problem are found](https://github.com/sintaxi/surge/issues/new).

**Comment by raygesualdo** [comment]
Can we encrypt wildcards as well? For example, if I set as a custom domain `mycompany.com` and want to publish to both the bare domain and N number of subdomains, can I encrypt `*.mycompany.com` or similar?
